### PR TITLE
Feat/comment editor

### PR DESCRIPTION
**Pull request by DarianRushworth**
**Comments editable**
Comments made by the specialist are immediately rendered, with all the proper information, all I need to do was tinker with the **useEffect** call. I also placed a icon for the user to be able to edit his or her comment. Upon editing the same form appears with previous data as placeholders. The comment is made and rendered immediately. I also then made sure that only the user that posted the comment can edit it.

### CODE REVIEW

File: src/components/SpecialistPortal/Dashboard/SpecialistDashboard.js [hunk 1]
```diff
@@ -5,13 +5,15 @@ import { Redirect, Link } from "react-router-dom";
 import "./SpecialistDashboard.css";
 import styled from "@emotion/styled";
 import Card from "@material-ui/core/Card";
+import Spinner from "../../reogranisation/Spinner";
 // import assess from '../../../res/assess-white.png'
 // import invest from '../../../res/invest-white.png'
 // import crowdfunding from '../../../res/crowdfunding.png'
```
Comment (dashakrolik) [inline src/components/SpecialistPortal/Dashboard/SpecialistDashboard.js:11]: delete unused code

File: src/components/SpecialistPortal/Dashboard/CommentSection/CommentSection.js [hunk 2]
```diff
@@ -9,35 +9,45 @@ import CommentForm from "./CommentForm";
...
   const [commentsData, setCommentsData] = useState([]);
   const [showAddComment, setShowAddComment] = useState(false);
-
-  useEffect(() => {
+  
+  const fetchComments = () => {
     request
       .get(`${baseUrl}/ideas/${id}/comments`)
       .set("Authorization", `Bearer ${props.authState.token}`)
       //   .then((res) => console.log("res.body", res.body));
```
Comment (dashakrolik) [inline src/components/SpecialistPortal/Dashboard/CommentSection/CommentSection.js:20]: delete unused

**Review by dashakrolik — CHANGES_REQUESTED** [review]

**Review by mr-robak — APPROVED** [review]
looks good!

File: src/components/SpecialistPortal/Dashboard/SpecialistDashboard.js (hunk 1 above)
Comment (DarianRushworth) [reply src/components/SpecialistPortal/Dashboard/SpecialistDashboard.js]: Code deleted, see below for commit.

File: src/components/SpecialistPortal/Dashboard/CommentSection/CommentSection.js (hunk 2 above)
Comment (DarianRushworth) [reply src/components/SpecialistPortal/Dashboard/CommentSection/CommentSection.js]: Code deleted.

**Review by nazneen1022 — APPROVED** [review]
looks good.

**Review by dashakrolik — APPROVED** [review]

**Comment by dashakrolik** [comment]
Solve merge conflicts and good to merge!

**Pull request merged**